plugins/lsp/conform-nvim.nix

```diff
@@ -0,0 +1,111 @@
+        "see documentation"
+        ''
+          # Map of filetype to formatters
+          formattersByFt =
+            {
+              lua = [ "stylua" ];
+              # Conform will run multiple formatters sequentially
+              python = [ "isort" "black" ];
+              # Use a sub-list to run only the first available formatter
+              javascript = [ [ "prettierd" "prettier" ] ];
+              # Use the "*" filetype to run formatters on all filetypes.
+              "*" = [ "codespell" ];
+              # Use the "_" filetype to run formatters on filetypes that don't
+              # have other formatters configured.
+              "_" = [ "trim_whitespace" ];
+             };
+        '';
+
+      formatOnSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+            timeoutMs =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter on save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+          See :help conform.format for details.
+        '';
+
+      formatAfterSave =
+        helpers.defaultNullOpts.mkNullable (types.submodule {
+          options = {
+            lspFallback =
+              mkOption
+              {
+                type = types.bool;
+                description = "See :help conform.format for details.";
+              };
+          };
+        })
+        "see documentation"
+        ''
+          If this is set, Conform will run the formatter asynchronously after save.
+          It will pass the table to conform.format().
+          This can also be a function that returns the table.
+        '';
+
+      # Set the log level. Use `:ConformInfo` to see the location of the log file.
+      logLevel = helpers.defaultNullOpts.mkEnumFirstDefault ["ERROR" "DEBUG" "INFO" "TRACE" "WARN" "OFF"] "Set the log level. Use `:ConformInfo` to see the location of the log file.";
+
+      # Conform will notify you when a formatter errors
+      notifyOnError =
+        helpers.defaultNullOpts.mkBool true;
+
+      # Custom formatters and changes to built-in formatters
+      formatters =
+        helpers.defaultNullOpts.mkNullable (types.attrsOf types.anything)
+        "see documentation"
+        "Custom formatters and changes to built-in formatters";
+    };
+
+  config = let
+    setupOptions = with cfg;
+      {
+        inherit formattersByFt;
+        log_level = helpers.ifNonNull' cfg.logLevel (helpers.mkRaw "vim.log.level.${cfg.logLevel}");
```

Comment:
you will need to do `formatters_by_ft = formmatersByFt;` as the plugins expects a snake case.
You also need to add here the other options like `format_on_save`, `notify_on_error`, ....